### ISSUE TITLE
ERROR: export 'useInsertionEffect' (imported as 'e') was not found in 'react'

### ISSUE THREAD

**Issue by pnob32** [ndom91/react-timezone-select#87]
### Describe the bug

I'm seeing this error when building with webpack, React 17, and esbuild-loader.

```
ERROR in ./node_modules/react-timezone-select/dist/esm/index.js 9:10466-10486
export 'useInsertionEffect' (imported as 'e') was not found in 'react'
```

I am building my package with `type: "commonjs"` 

### Reproduction

building the module reproduces the issue: `npm run webpack`

### System Info

```Shell
OS: Linux 5.4
Firefox: 102.9.0esr
```

### Used Package Manager

npm

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by jhardin293** [comment]
+1 Same error
- React 17
- Next 12

**Comment by ndom91** [comment]
Hmm, this doesn't ring a bell. We're not purposely using this in the "user land" code of the component. 

We changed a good deal related to bundling and building yesterday. 

Can you give the latest version on npm another shot and let me know if you still have this issue?

**Comment by ndom91** [comment]
Should be fixed in the latest versions, please give `v2.1.0+` a shot :pray:
